src/main/java/com/javadiscord/jdi/internal/api/impl/scheduledevents/ModifyScheduledEventRequest.java

```diff
@@ -0,0 +1,66 @@
+package com.javadiscord.jdi.internal.api.impl.scheduledevents;
+
+import com.javadiscord.jdi.internal.api.DiscordRequest;
+import com.javadiscord.jdi.internal.api.DiscordRequestBuilder;
+import com.javadiscord.jdi.internal.models.guild.EntityMetadata;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/* NOTE TO FUTURE DEVS WHEN WE IMPLEMENT HIGHER LEVEL API
+If updating entity_type to EXTERNAL:
+
+    channel_id is required and must be set to null
+    entity_metadata with a entityMetadata field must be provided
+    scheduled_end_time must be provided
+ */
+public record ModifyScheduledEventRequest(
+        String guildId,
+        String scheduledEventId,
+        Optional<Long> channelId,
+        Optional<EntityMetadata> entityMetadata,
+        Optional<String> name,
+        Optional<Integer> privacyLevel,
+        Optional<Long> scheduledStartTime,
+        Optional<Long> scheduledEndTime,
+        Optional<String> description,
+        Optional<Integer> entityType,
+        Optional<Integer> status,
+        Optional<String> image)
+        implements DiscordRequest {
+    @Override
+    public DiscordRequestBuilder create() {
+        Map<String, Object> body = new HashMap<>();
+
+        // check if entityType = 3 (EXTERNAL)
+        // channelId must be set to null
+        // entityMetadata (entity_metadata), scheduledEndTime are all required
+        if (entityType.isPresent() && entityType.get() == 3) {
+            if (entityMetadata.isEmpty() || scheduledEndTime.isEmpty()) {
```

Comment:
Please can this if-check be moved into the constructor? Thanks

Reply:
We discussed on discord and have decided to shelf this until the higher level api is implemented. That validation will stay for now.